### ISSUE TITLE
add extra status verification before automation start

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/mcloud-grid#75]
to handle correctly al kind of states described here: https://github.com/zebrunner/stf/issues/483
we have to add explicit verification of the status: 3

**Comment by vdelendik** [comment]
@dhreben, closing as won't fix but we have to test explicitly below idea:
1. let's move device to unhealthy state
2. launch mobile job explicitly providing this problematic device
-> based on my understanding we verify that device is available (getPresent) and available for usage (getReady): https://github.com/zebrunner/mcloud-grid/blob/221ad41c8bb3e098dc6bc823f2e702781d8bfa1d/src/main/java/com/zebrunner/mcloud/grid/integration/client/STFClient.java#L137
so if device in state unhealthy - 3 we shouldn't proceed with test and request will stuck in queue

**Comment by dhreben** [comment]
It will be verified after the execution https://github.com/zebrunner/stf/issues/825

**Comment by vdelendik** [comment]
@akamarouski, please take a look how we verify device state and make sure we don't use device in unhealthy state (number 3?!)

**Comment by akamarouski** [comment]
Added STF device status verification for:

> If status is null, we will show `[STF-%s] STF device status is null. It will be ignored: %s minutes.`

>If status is `2` (UNAUTHORIZED), we will show `[STF-%s] STF device status 'UNAUTHORIZED'. It will be ignored: %s minutes.`

>If status is `7` (UNHEALTHY), we will show `[STF-%s] STF device status 'UNHEALTHY'. It will be ignored: %s minutes.`

Also we do not want to ask STF for this device when we got such statuses for each session request. We will use timeouts and during this timeout we will not use STF api for this device. We will see for example such message in logs:

 `[STF-%s] The next attempt to reserve '%s' STF device will be given only after '4' minutes."`
 
 Default timeouts:
 `status null` - `10 minutes` - override using `STF_DEVICE_INVALID_RESPONSE_IGNORE_TIMEOUT` env variable
 `UNAUTHORIZED` - `10 minutes` - override using `STF_DEVICE_UNAUTHORIZED_IGNORE_TIMEOUT` env variable
  `UNHEALTHY ` - `5 minutes` - override using `STF_DEVICE_UNHEALTHY_IGNORE_TIMEOUT` env variable

**Comment by akamarouski** [comment]
Checked `UNHEALTHY` device status. Could not test UNAUTHORIZED because when device it this state, appium container could not be registered in the grid.

**Comment by dhreben** [comment]
UNHEALTHY is tested 
status null and UNAUTHORIZED will be tested on production
